src/cycax/cycad/engine.py

```diff
@@ -1,13 +1,217 @@
+            return "rotate([" + str(0) + ", " + str(90) + ", " + str(0) + "])"
+        elif side == RIGHT:
+            return "rotate([" + str(0) + ", " + str(270) + ", " + str(0) + "])"
+        elif side == FRONT:
+            return "rotate([" + str(270) + ", " + str(0) + ", " + str(0) + "])"
+        elif side == BACK:
+            return "rotate([" + str(90) + ", " + str(0) + ", " + str(0) + "])"
+        else:
+            msg = f"Side: {side} is not one of TOP, BOTTOM, LEFT, RIGHT, FRONT, BACK."
+            raise ValueError(msg)
+
+
+    def decode(self, data_file: str):
+        """
+        This is the main working class for decoding the scad. It is necessary for it to be refactored.
+        
+        !!!For this method to work properly it will be necessary to add a JSON, STL and SCAD file into the working repository.!!!
+        
+        Args: 
+            data_file : name of the file that is to be decoded into a scad.
+        """
+        out_name = "./SCAD/" + data_file + ".scad"
+        SCAD = open(out_name, "w")
+        in_name = "./JSON/" + data_file + ".json"
+        
+        with open(in_name) as f:
+            data = json.load(f)
+
+        output = []
+        dif = 0
+        for action in data:
+            if action["type"] == "cut":
+                dif = dif + 1
+                output.insert(0, self.decode_cut())
+
+            output.append(self.translate(action))
+
+
+            if action["side"] != None:
+                output.append(self.rotate(action["side"]))
+
+            if action["name"] == "cube":
+                output.append(self.decode_cube(action))
+
+            if action["name"] == "external":
+                output.append(self.decode_external(data_file))
+
+            if action["name"] == "hole":
+                output.append(self.decode_hole(action))
+
+            if action["name"] == "nut":
+                output.append(self.decode_nut(action))
+
+        i = 0
+        while i < dif:
+            i = i + 1
+            output.append("}")
+
+        for out in output:
+            SCAD.write(out)
+
+        SCAD.close()
+
+    def render_stl(self, filename: str):
+        """
+        
+        This takes a SCAD object and runs a command through linex that converts it into an stl. 
+        Depending on the complexity of the object it can take long to compute.
+        It prints out some messages to the terminal so that the impatient user will hopefully wait.(Similar to many windows request.)
+        
+        Args:
+            filename : This is the name of the file which will be converted from a scad to a stl
+        
+        """
+        out_name = "./SCAD/" + filename + ".scad"
+        out_stl_name = "./STL/" + filename + ".stl"
 
+        print("!!! THIS WILL TAKE SOME TIME, BE PATIENT !!!")
+        result = subprocess.run(
+            ["openscad", "-o", out_stl_name, out_name], capture_output=True, text=True
+        )
+        print("stdout:", result.stdout)
```

Comment:
Consider using logging. e.g. Especially for a library.

```
import logging
if result.stdout:
    logging.info('OpenSCAD: %s', result.stdout)
if result.stderr:
    logging.error('OpenSCAD: %s'. result.stderr)
```
Some time existence of stderr is not an error then you need to look at return code to know if its and error, warn, or info.